Library/Homebrew/cmd/list.rb

```diff
@@ -86,7 +86,13 @@ def list
         ls_args << "-r" if args.r?
         ls_args << "-t" if args.t?
 
-        safe_system "ls", *ls_args, HOMEBREW_CELLAR
+        if !$stdout.tty? && !args.formula?
+          odeprecated "`brew list` to only list formulae", "`brew list --formula`"
+          safe_system "ls", *ls_args, HOMEBREW_CELLAR
+        else
+          safe_system "ls", *ls_args, HOMEBREW_CELLAR
```

Comment:
This will always print formulae, even if *only* `--cask` is passed.

Reply:
I have thought that it does not go after this line of code: https://github.com/Homebrew/brew/blob/f84b9a02753e240b4701a6969d0f912c32e61240/Library/Homebrew/cmd/list.rb#L60

Reply:
You're right.